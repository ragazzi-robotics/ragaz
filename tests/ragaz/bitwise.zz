def main():
    var x = 3

    print(~x)
    print(x << 2)
    print(x >> 1)
    print(x & 1)
    print(x | 4)
    print(x ^ 1)

    x = 3
    x <<= 2
    print(x)

    x = 3
    x >>= 1
    print(x)

    x = 3
    x &= 2
    print(x)

    x = 3
    x |= 4
    print(x)

    x = 3
    x ^= 1
    print(x)
