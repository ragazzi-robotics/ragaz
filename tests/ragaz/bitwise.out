-4
12
1
1
7
2
12
1
2
7
2
